api/v1alpha2/conversion_test.go

```diff
@@ -0,0 +1,22 @@
+package v1alpha2
+
+import (
+	"testing"
+
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api-provider-vsphere/api/v1alpha3"
+)
+
+func TestFuzzyConversion(t *testing.T) {
+	g := NewWithT(t)
+	scheme := runtime.NewScheme()
+	g.Expect(AddToScheme(scheme)).To(Succeed())
+	g.Expect(v1alpha3.AddToScheme(scheme)).To(Succeed())
+
+	//t.Run("for VSphereCluster", utilconversion.FuzzTestFunc(scheme, &v1alpha3.VSphereCluster{}, &VSphereCluster{}))
+	t.Run("for VSphereMachine", utilconversion.FuzzTestFunc(scheme, &v1alpha3.VSphereMachine{}, &VSphereMachine{}))
```

Comment:
Still sorting out the `VSphereCluster` and `VSphereMachineTemplate` manual conversions. Will uncomment as soon as it's done.

Reply:
The conversion test for `VSphereCluster` is still failing.  Both objects _seem_ to be identical and the [ObjectDiff](https://github.com/kubernetes-sigs/cluster-api/blob/master/util/conversion/conversion.go#L154) is empty.

Reply:
Just to update - Had a helpful discussion with @vincepri  (thanks!). The tests are failing due to the [conversion scope issue](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/749#discussion_r381545611).